Develop #96 rework countdown endpoint

davkra opened the pull request:
Closes #96 

# 💻 What
Reworked countdown endpoint as only a single countdown must be present at all time in the database

# ⚠️ Attention!

<del>Database table `countdown` needs to be deleted potentially, which must be done by the DevOp Team!</del>
Migration files are added and deplyoment should work automatically!

davkra committed:
rework countdown

Only a single countdown must be present at all time in the database.

davkra committed:
change countdown table in hindsight

max-pfi commented on src/main/resources/db/81-create-countdown-table.xml:
Im Grunde müsstest du jetzt die gesamte DB löschen - das migration File ist in der DB auf Test schon durchgelaufen und ich denke nicht, dass changes im Nachhinein noch irgendeine Wirkung haben, weil die DB speichert welche Migrations schon fertig sind.

Ideal wäre ein neues Migration File zum Löschen vom Table + ein weiteres Migration File um den Table zu erstellen. Alternativ eben die gesamte DB auf Test droppen - auf Prod waren die Changes soweit ich weiß noch nie, hier würden sie also zum Ersten Mal und damit korrekt laufen.

max-pfi commented on src/main/java/at/fh/joanneum/irfc/rest/CountdownApi.java:
```diff
@@ -4,63 +4,34 @@
-    @POST
-    @Produces(MediaType.APPLICATION_JSON)
-    @Consumes(MediaType.APPLICATION_JSON)
     @Authenticated
```
Sollte soweit ich weiß öffentlich sein und keine Authentication verlangen.

max-pfi reviewed (approved):
Grundsätzlich funktioniert alles, wobei die Changes aus dem XML File wie im Comment beschrieben nicht übernommen wierden, wenn du nur den Table löschst - in dem Fall wird er auch nicht neu erstellt und fehlt. Also ich würde sagen entweder wie beschrieben neue Migrations-Files oder die gesamte DB (mit Volume) löschen. Falls wir es schon auf Prod haben (was ich nicht glaube) müssten wir auf jeden Fall mit neuen Migration-Files arbeiten. 
Sonst noch den einen get endpoint public machen und mit den Änderungen gerne mergen.

davkra committed:
Revert "change countdown table in hindsight"

This reverts commit e7030bbda3df272bf5197a5c84acb520c4080cf2.

Do not change migration files in hindsight!

davkra committed:
change countdown table

Only one entry is needed, as there will only be a single date stored for the countdown.

davkra committed:
removed authentication for countdown api

davkra commented:
Neues Migration File wurde angelegt und die Änderungen so angepasst, damit nur noch ein einzelner Countdown anzufinden ist. 

Merge sollte jedoch zuerst mit dem Webapp team abgesprochen werden, damit diese den (einzelnen) Datumswert richtig auslesen.
